megamek/src/megamek/client/ui/WrapLayout.java

```diff
@@ -0,0 +1,194 @@
+    /**
+     * Constructs a new <code>WrapLayout</code> with a left
+     * alignment and a default 5-unit horizontal and vertical gap.
+     */
+    public WrapLayout() {
+        super();
+    }
+
+    /**
+     * Constructs a new <code>FlowLayout</code> with the specified
+     * alignment and a default 5-unit horizontal and vertical gap.
+     * The value of the alignment argument must be one of
+     * <code>WrapLayout</code>, <code>WrapLayout</code>,
+     * or <code>WrapLayout</code>.
+     * @param align the alignment value
+     */
+    public WrapLayout(int align) {
+        super(align);
+    }
+
+    /**
+     * Creates a new flow layout manager with the indicated alignment
+     * and the indicated horizontal and vertical gaps.
+     * <p>
+     * The value of the alignment argument must be one of
+     * <code>WrapLayout</code>, <code>WrapLayout</code>,
+     * or <code>WrapLayout</code>.
+     * @param align the alignment value
+     * @param hgap the horizontal gap between components
+     * @param vgap the vertical gap between components
+     */
+    public WrapLayout(int align, int hgap, int vgap) {
+        super(align, hgap, vgap);
+    }
+
+    /**
+     * Returns the preferred dimensions for this layout given the
+     * <i>visible</i> components in the specified target container.
+     * @param target the component which needs to be laid out
+     * @return the preferred dimensions to lay out the
+     * subcomponents of the specified container
+     */
+    @Override
+    public Dimension preferredLayoutSize(Container target) {
+        return layoutSize(target, true);
+    }
+
+    /**
+     * Returns the minimum dimensions needed to layout the <i>visible</i>
+     * components contained in the specified target container.
+     * @param target the component which needs to be laid out
+     * @return the minimum dimensions to lay out the
+     * subcomponents of the specified container
+     */
+    @Override
+    public Dimension minimumLayoutSize(Container target) {
+        Dimension minimum = layoutSize(target, false);
+        minimum.width -= (getHgap() + 1);
+        return minimum;
+    }
+
+    /**
+     * Returns the minimum or preferred dimension needed to layout the target
+     * container.
+     *
+     * @param target target to get layout size for
+     * @param preferred should preferred size be calculated
+     * @return the dimension to layout the target container
+     */
+    private Dimension layoutSize(Container target, boolean preferred) {
+        synchronized (target.getTreeLock()) {
+            // Each row must fit with the width allocated to the container.
+            // When the container width = 0, the preferred width of the container
+            // has not yet been calculated so lets ask for the maximum.
+
+            int targetWidth;
+            Container container = target;
+
+            while ((container.getSize().width) == 0 && (container.getParent() != null)) {
```

Comment:
Just noticed the first brackets are around the first parameter and not the comparison 😂

Reply:
Hehe I had to look at it for a while to see it :D. Corrected now. Hopefully...